phenopy/d2p.py

```diff
@@ -86,10 +54,88 @@ def load(phenotype_annotations_file, hpo_network, alt2prim, logger=None):
+        if logger is not None:
+            logger.critical(hpoa_file_error_msg)
+        else:
+            sys.stderr.write(hpoa_file_error_msg)
+        exit(1)
+
+
+def load(phenotype_annotations_file, hpo_network, alt2prim, default_frequency=0.5):
+    """Parse the hpoa file
+    :param phenotype_annotations_file: path to the phenotype.hpoa file
+    :return: three dictionaries of disease to phenotypes, phenotypes to disease, and phenotypes to disease frequencies
+    """
+    if phenotype_annotations_file.endswith("hpoa"):
+        records = read_hpo_annotation_file(phenotype_annotations_file,hpo_network)
+    else:
+        records = read_custom_annotation_file(phenotype_annotations_file,hpo_network)
+
+    disease_to_phenotypes = dict()
+    phenotype_to_diseases = dict()
+
+    for r in records:
+        term_id, disease_accession, freq = r
+        if term_id not in phenotype_to_diseases:
+            phenotype_to_diseases[term_id] = {disease_accession: {'frequency': default_frequency}}
+        else:
+            if disease_accession not in phenotype_to_diseases[term_id]:
+                phenotype_to_diseases[term_id].update({disease_accession: {'frequency': default_frequency}})
+
+        phenotype_to_diseases[term_id][disease_accession]['frequency'] = freq
+
+        # add the phenotype to the disease in the disease_records dictionary
+        if disease_accession not in disease_to_phenotypes:
+            disease_to_phenotypes[disease_accession] = {'record_id': disease_accession,
+                                                        'terms': [],
+                                                        'weights': {'disease_frequency': [],
+                                                                    },
+                                                        }
+        disease_to_phenotypes[disease_accession]['terms'].append(term_id)
+
+    # going from dict to a list of disease records and setting weights
+    disease_records = list()
+    for disease_accession, disease in disease_to_phenotypes.items():
+        disease['terms'] = sorted(set(disease['terms']))
+        for term_id in disease['terms']:
+            # convert alternate phenotype id to primary
+            term_id = term_id if term_id not in alt2prim else alt2prim[term_id]
+            if term_id not in hpo_network.nodes():
+                continue
+
+            frequency_weight = phenotype_to_diseases[term_id][disease_accession]['frequency']
+            #
+            disease['weights']['disease_frequency'].append(frequency_weight)
+
+        disease_records.append(disease)
+
+    # TODO: do we need phenotype_to_diseases?
+    return disease_records, phenotype_to_diseases
+
+
 def frequency_converter(hpoa_frequency):
     """convert the frequency column from the hpoa file to a float"""
     if 'HP:' in hpoa_frequency:
-        return hpo_id_to_float.get(hpoa_frequency, '')
+        return hpo_id_to_float.get(hpoa_frequency, 0.5)
```

Comment:
If no frequency is given, we assume it's half? That seems strong. Can you put a `#TODO` here so we can come back and discuss the best default?